src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -351,34 +348,39 @@ public void doGenerateAllBundles(StaplerRequest2 req, StaplerResponse2 rsp) thro
      * @throws IOException If an input or output exception occurs
      */
     @RequirePOST
-    public HttpRedirect doGenerateAllBundlesAsync(StaplerRequest2 req, StaplerResponse2 rsp)
+    public HttpRedirect doGenerateAllBundleAsync(StaplerRequest2 req, StaplerResponse2 rsp)
             throws ServletException, IOException {
         JSONObject json = req.getSubmittedForm();
         if (!json.has("components")) {
             rsp.sendError(SC_BAD_REQUEST);
             return new HttpRedirect("support");
         }
         final List<Component> components = getComponents(req, json);
+        UUID taskId = UUID.randomUUID();
 
-        for (Component component : components) {
-            if (component instanceof AboutBrowser) {
-                AboutBrowser aboutBrowser = (AboutBrowser) component;
-                aboutBrowser.setScreenResolution(Functions.getScreenResolution());
-                aboutBrowser.setCurrentRequest(Stapler.getCurrentRequest2());
-                aboutBrowser.setGeneratedAsync(true);
+        // There are some components that need the request components to be processed
+        // these components cannot be processed async
+        // so process them first and then process the other components async
+        List<Component> syncComponent =
+                components.stream().filter(c -> !c.canBeGeneratedAsync()).toList();
+        if (!syncComponent.isEmpty()) {
+            File outputDir = new File(SUPPORT_BUNDLE_CREATION_FOLDER + "/" + taskId);
```

Comment:
Maybe easier to create `support-bundle-$taskId.zip` directly in `SUPPORT_BUNDLE_CREATION_FOLDER` rather than needing to create and clean up a subdirectory.